src/odemis/dataio/test/tiff_test.py

```diff
@@ -2729,6 +2732,65 @@ def test_convert_openfibsem_to_odemis_metadata(self):
             else:
                 self.assertEqual(value, data.content[0].metadata[key])
 
+PYRAMID_FILENAME = "test-pyramid.ome.tiff"
+class TestTileRead(unittest.TestCase):
+
+    def tearDown(self):
+        # clean up
+        try:
+            os.remove(PYRAMID_FILENAME)
+        except Exception:
+            pass
+
+    def test_tiled_read(self):
+        """Test the tiled read functions"""
+
+        # storing full metadata for completeness
+
+        md = {
+            model.MD_PIXEL_SIZE: (7.738095238095199e-08, 7.738095238095199e-08),
+            model.MD_POS: (0.051781573195293, -0.007091270569792),
+            model.MD_ACQ_DATE: 1727296375,
+            model.MD_ROTATION: 6.1837053071795856,
+            model.MD_SHEAR: -4.3021142204224836e-16,
+            model.MD_DIMS: "YX",
+        }
+
+        # create a dummy overview pyramid with the same problematic shape
+        das = model.DataArray(
+            numpy.zeros((8194, 7143), dtype=numpy.uint16), metadata=md
+        )
+
+        # export the dummy overview pyramid
+        export(PYRAMID_FILENAME, das, pyramid=True)
+
+        # open the pyramid and create a projection
+        data = open_acquisition(PYRAMID_FILENAME)
+        stream = data_to_static_streams(data)[0]
+        proj = RGBSpatialProjection(stream)
+        proj._getTilesFromSelectedArea()
```

Comment:
What are you testing here? Is it just that _getTilesFromSelectedArea() doesn't fail? It's odd as it's a private function. If so, you should at least add a comment, and otherwise, try to trigger that call via an interaction using the public API.

Also, if this is to test the projection, this should go to src/odemis/acq/test/stream_test.py

Reply:
This is to test the tiled read still works correctly after updating the image. I think it makes more sense here, because it actually tests the tile-read from disk and the projection. the projection tests in stream_test use mocked tile reads so they dont pick up these errors.